src/wallet/rollover.rs

```diff
@@ -0,0 +1,453 @@
+use chrono;
+use curve25519_dalek::{
+    ristretto::CompressedRistretto, ristretto::RistrettoPoint, scalar::Scalar,
+    traits::MultiscalarMul,
+};
+use merlin::Transcript;
+use rand_core::{CryptoRng, RngCore};
+
+use crate::{constants, Epoch, EpochState, Tag};
+
+use super::keys::{Parameters, Secrets};
+use super::Wallet;
+
+mod proofs {
+    define_proof! {
+        client,
+        "WalletRollover_Client",
```

Comment:
Just noting this for later: this matches the docs and the other proofs, but looking at it again I think that it would be cleaner (in a future issue / PR) to change them all to `wallet::rollover::client` so that the domain separation strings match up with the Rust modules.

Reply:
https://github.com/hdevalence/danake/issues/19